### ISSUE TITLE
reponse.index changes depending of interaction

### ISSUE THREAD

**Issue by mikkhait** [Microsoft/BotBuilder#299]
This one was expensive one :)

Given `builder.Prompts.choice(session, "Ask Questions", ['Q1','Q2']`
When I press **Q1** as a button on _a device_ or _a button in the messenger client_
Then I receive `reponse.index` ===  0

Given `builder.Prompts.choice(session, "Ask Questions", ['Q1','Q2']`
When I enter `1` in a text field for **Q1**
**ACTUAL**
Then I receive `reponse.index` ===  1
**EXPECTED**
Then I receive `reponse.index` ===  0

```
{ resumed: 0,
  promptType: 3,
  response: { index: 0, entity: 'Score', score: 1 },
  childId: 'BotBuilder.Dialogs.Prompts' }

 { resumed: 0,
  promptType: 3,
  response: { index: 0, entity: 'Score', score: 1 },
  childId: 'BotBuilder.Dialogs.Prompts' }
```

**Comment by mikkhait** [comment]
Actually, I suppose for consistency sake, we should start index from `1`, since there is never an option for `0`

Thoughts?

**Comment by Stevenic** [comment]
I'll have to debug that.

**Comment by Stevenic** [comment]
Yep... It's a bug. I have a fix. Should be released tomorrow along with some other changes.

**Comment by mikkhait** [comment]
@Stevenic thanks a lot for the fast turnaround. Which way are you going with `1` or `0`?

**Comment by Stevenic** [comment]
It will return 0. That was my intention. The user types 1-to-_n_ but you should get back 0-to-_n_. I had done the _n_-1 to return the proper entity but for the index I was just returning _n_.  I'm completely re-writing the prompts right now to make some general improvements so found the bug while doing that.

**Comment by mikkhait** [comment]
fixed in 0.11.0. thank you
